### PR TITLE
[Seriously, it's ready]Virology Rework: The Thresholdening

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
experiment: Viruses and symptoms have been havily reworked.
add: Symptoms now have statistic thresholds, that give them new properties or improve their existing ones if the overall virus statistic is above the threshold. Check the pull request in github or the wiki (soon) for the full list.
del: Some symptoms no longer scale linearly with stats, and instead have thresholds.
tweak: The symptom limit is now 6.
del: Viruses can no longer be made invisible to the Pandemic
tweak: Symptoms no longer trigger with a 5% chance every second, but instead have a minimum and maximum number of seconds between each activation, making them more consistent.
del: The symptoms Blood Vomit and Projectile Vomit have been removed, and are now bonuses for the base Vomit symptom.
del: The Weakness symptom has been removed as it was completely useless.
tweak: The Sensory Destruction symptom has been reworked into Narcolepsy, which causes drowsiness and sleep.
tweak: Viral Aggressive Metabolism now has a timer before it starts decaying the virus. It scales with the highest between Resistance or Stage Speed.
add: You can now neuter symptoms, making them inactive. They will still affect stats. Adding formaldehyde to a virus will neuter a random symptom. A bottle of formaldehyde starts in the virus fridge.
/:cl:

Full list of threshold bonuses:
* Facial Hypertrichosis - None
* Choking 
  * Stage Speed 8 : Acts more often
  * Stealth 4: Suppresses early warnings
* Acute respiratory distress syndrome
  * Stage Speed 8: Adds paralyzing chems (removed from base version)
  * Infect 8: Stronger damage
* Confusion
  * Stealth 4: Suppresses early warnings
  * Resistance 6: Also deals some brain damage, not above 80
  * Infect 6: Stronger confusion
* Cough
  * If AIRBORNE: Coughing causes a spread at range 1
  * Resistance 3: Coughing drops small items (removed from base)
  * Resistance 10: Can cause coughing fits that stun briefly
  * Speed 6: Increases cough frequency
* Deafness
  * Stealth 4: Suppresses early warnings
  * Resist 9: Deafness is permanent instead of temporary
* Dizziness
  * Stealth 4: Suppresses early warnings
  * Infect 6: Also adds the druggy effect
* Fever
  * Resist 5: Heats more, can go above the damage threshold
  * Resist 10: Heats even more per activation
* Shiver
  * Speed 5: Cools more, can go below the damage threshold
  * Speed 10: Cools even more
* Spontaneous Combustion
  * Stealth 4: Suppresses early warnings
  * Speed 5: Increases fire stacks and damage
  * Speed 10: Increases fire_stacks and damage even more
  * Infect 8: Spreads the virus when igniting
* Alkali Perspiration
  * Resist 9: Increases time between ignitions but increases power as well
  * Speed 8: Doubles the explosion range when wet
  * Infect 8: Also adds napalm and clf3 (removed from base version)
* Necrotizing Fasciitis
  * Resist 7: Causes extra bleeding
  * Infect 8: Deals additional stamina damage
* Autophagocytosis Necrosis
  * Stealth 5: Suppresses early warnings
  * Speed 7: Adds Heparin and Lipolicide (removed from base version)
  * [PROPOSED REQUESTING APPROVAL] Infect 13: Adds romerol, turning the victim into a zombie on death.
* DNA Saboteur
  * Stealth 5: DNA won't reset after curing
  * Speed 10: Increases mutation frequency
  * Resist 8: Mutate twice instead of once
* Hallucigen
  * Stealth 4: Suppresses early warnings, and gives off the messages of positive symptoms instead
  * Speed 7: Stronger hallucinations
* Headache
  * Stealth 4: Reduces early message frequency
  * Speed 6: Deals some stamina damage when headaching
  * Speed 9: Reduces frequency, headaches are strong enough to stun
* Healing Symptoms
  * Stealth 4: Hides the overlay when healing
  * Speed 7: Increases healing rate
  * Speed 15: Further increases healing rate
* Itching
  * Infect 6: Increases frequency
  * Speed 7: Itching causes scratching, dealing minor brute damage (0.5)
* Narcolepsy
  * Infect 7: Deals stamina damage when it triggers
  * Resistance 10: Triggers more often
* Self-Respiration
  * Resist 8: Regenerates blood
* Sensory Restoration
  * Resist 6: Also heals brain damage (removed from base)
  * Infect 8: Also purges alcohol (removed from base)
* Sneezing
  * Infect 9: Increases spread range
  * Stealth 4: Suppresses early warnings
* Hyphema
  * Stealth 4: Suppresses early warnings
  * [REQUESTING APPROVAL] Resist 13: Eyes fall off instead of just being blinded
* Voice Change
  * Stealth 3: Suppresses early warnings
  * Speed 7: Changes voice fater
  * [REQUESTING APPROVAL] Infect 14: Changes the infectee's language along with the voice
* Vomit
  * Stealth 4: Suppresses early warnings
  * Resist 7: Blood Vomit
  * Infect 7: Projectile Vomit

**Unlisted symptoms do not have threshold bonuses!**

All balance is not final, i'll tweak the numbers as necessary.

## Why
Threshold bonuses: The reasoning behind these is that they feel more rewarding and impactful than simple number scaling, and gives the option to dedicate symptom space to more symptoms or to upgrade a few central ones instead.
Symptom activation change: the intention behind this is RNG consistency and ease of balance. Currently all symptoms have a 5% chance of activate with few exceptions, making it complicated to tweak.
Neutering: Upgrading positive symptoms by putting a ton of negative ones doesn't work well, in my opinion, especially when they'll trigger their own thresholds making the virus somewhat dangerous instead. Neutering allows players to sidestep that, but i recognize that it swings a bit too far on the other side of the risk/reward balance. I plan on finding a better system to manage virus statistics in the future.

### CODE REVIEW

**Comment by kevinz000** [comment]
6 symptoms 3 viruses again or 6 symptoms one virus because the latter is REALLY weak.

**Comment by DoctoreMad** [comment]
>[PROPOSED REQUESTING APPROVAL] Infect 13: Adds romerol, turning the victim into a zombie on death.

I SERIOUSLY HOPE THIS ISN'T APPROVED MYSELF

**Comment by XDTM** [comment]
@kevinz000 Overall symptoms are stronger, i think, and i really want to err on the side of underpowered rather than virology killing everyone

**Comment by kevinz000** [comment]
damnit rip my healing virus cheese electric boogaloo :'(

**Comment by Armhulen** [comment]
doctor i know you hate fun but just work with me here

**Comment by DoctoreMad** [comment]
Can you tune it down to normal non-antag zombies instead of a TC item? Looks like powercreep to me.

**Comment by PKPenguin321** [comment]
Zombies miiiight be sketchy but personally I like it

Other two things you want approval on I think should definitely go through they're hilarious

**Comment by XDTM** [comment]
lzimann was very against zombies, i dunno if maintainer or headmin ranks higher here

**Comment by PKPenguin321** [comment]
He's not the only maintainer you know

**Comment by Shadowlight213** [comment]
\> Neutering lets you get the stat boosts from bad symptoms while not having the symptom itself

No. Just no. This is just a flat out buff to heal viruses at this point.

**Comment by XDTM** [comment]
Two less max symptoms, you ain't reaching thresholds if you put in all 4 healing symptoms.

**Comment by Nabski89** [comment]
Did you change it so that Transmission doesn't subtract one for each symptom? If not the only ones of these that you can actually get are on vomiting and coughing.

Resistance: Hyphema 13 is unreachable.

Stealth: All of these work, however the nature of stealth means they will likely be even more frustrating than normal. You can't see what's in it, you can see what the cure is, and you can't even use all the symptoms to math it out. Stealth spreading is the new death plagues.

Speed: Spontaneous 10 is unreachable. All the healing at 15 is unreachable.

So many of these you'll have to blank out symptoms every time, that or they'll be on no matter what.

For example on stage speed to max it you'll want Accel, Youth, Fever, Itching, (Shivering or Headache)
This gives speed 17 with resist 11 or 9 (modified by whatever you mix with it).
This activates Fever, Itching, and (Shivering/Headache) all at once, in addition to whatever you add.

The activation is a great change, but either the activation numbers, the formulas, or viral numbers, need some tweaking.

**Comment by XDTM** [comment]
Thanks, i was sure i botched some numbers. As for the last point, it's the reason i added neutering; you'd have to add neutered fever etc to the healing symptom.

**Comment by Nabski89** [comment]
I struggle at reading code. Is the neutering targeted or more RNG hell?

If you can neuter symptoms what's the point of having the Viral Accel/Viral Self-Adaptation/Eternal Youth since these are all just effectless things used for their stats already and a mandatory add for the stat you want (speed/stealth/resist) {sneezing for transmission but it has an activated}.

Would you have any issue with the symptoms on virus's being revisited in a later PR to spread out the stat buffs from tier one?

Also this is a good system for balancing, as negative thresholds could be used to keep things in check as well.

**Comment by XDTM** [comment]
As i mentioned in the PR body neutering is far from a perfect system, and i'm going to look for a better way to handle stats. As for how it's obtained, it's picked at random when dropping formaldehyde, so if you isolate fever and formaldehyde it it's a 100% certain neutered fever.

I'm absolutely open to rebalancing the stats on symptoms, but i feel like stats should not be determined by the symptoms in the first place, as it makes virus builds way too rigid. Maybe there should be a point system where you could distribute x points to various stats, increasing with upgrades/research/whatever.

**Comment by XDTM** [comment]
oh what the hell

**Comment by XDTM** [comment]
how do i undo this shit

**Comment by XDTM** [comment]
what
why is it closed

**Comment by XDTM** [comment]
why can't i reopen it

**Comment by XDTM** [comment]
oh thank god

**Comment by Armhulen** [comment]
ok can we merge this? i just spent 35 minutes making a basic heal virus because of BYOND rng(tm)

**Comment by XDTM** [comment]
@Armhulen https://tgstation13.org/phpBB/viewtopic.php?f=2&t=11597

**Comment by Armhulen** [comment]
i followed the other guide i was just insanely unlucky

**Comment by XDTM** [comment]
New change: high stealth will no longer hide from the Pandemic. My reasoning for this is that it made viruses virtually incurable, and with this rework stealth is getting buffs by reducing the amount of warnings for the victims.

**Comment by Armhulen** [comment]
Ok great now MERGE IT

File: code/datums/diseases/advance/symptoms/vomit.dm
Comment (lordpidey) [inline code/datums/diseases/advance/symptoms/vomit.dm]: Perhaps we should uncapitalize the proc name for consistency?

File: code/datums/diseases/advance/symptoms/vomit.dm
Comment (lordpidey) [inline code/datums/diseases/advance/symptoms/vomit.dm]: Perhaps have this be an integer rather than a boolean, as it's something that can be scaled.

**Comment by Daturix** [comment]
I will find some way to meme lavawalking again

**Comment by Nabski89** [comment]
I managed to get the "Basic Healing (does nothing)" symptom when this was testmerged just by adding a bunch of virus food.

**Comment by XDTM** [comment]
who coded it to go from 0 to 2 seriously

**Comment by Nabski89** [comment]
I'm also not sure if the basic healing thing was healing anything.

**Comment by XDTM** [comment]
I thought the (does nothing) was a fairly straightforward clue about that

It's just a base class for all healing symptoms, that shouldn't be obtainable and has no effect.

File: code/datums/diseases/advance/symptoms/heal.dm
Comment (lordpidey) [inline code/datums/diseases/advance/symptoms/heal.dm]: replace the if statement with `hide_healing = A.properties["stealth"]>=4`

File: code/datums/diseases/advance/symptoms/itching.dm
Comment (lordpidey) [inline code/datums/diseases/advance/symptoms/itching.dm]: scratch = A.properties["stage_rate"]>=7

**Review by lordpidey — COMMENTED** [review]
Lots of these statements could be simplified, I just marked two as an example.

**Comment by XDTM** [comment]
If there's no big performance difference i'd like to keep these as is for code clarity

**Comment by chickenboy10** [comment]
Sounds great, I especially like the consistency in virus activation going up over time instead of being rng and  the pandemic seeing all viruses.
I also like the idea of viruses getting bonuses from certain stats, but I fear this will be a feature that is only available to people with the wiki open or lots of experience. It might end up being a change that makes virology needlessly over complex for little gain for most virologists.
Also, does the pandemic show any of the threshold bonuses you currently have active, as well as what symptoms are neutered?

**Comment by XDTM** [comment]
It does show neutered symptoms, but thresholds are still undocumented. I plan on adding being able to click on a symptom on the pandemic to see its details.

**Comment by Militaires** [comment]
threshold = cuckold??

**Comment by Nabski89** [comment]
There's some bad math going on in that post. You can't get 12 speed with three healing symptoms.

Accel +5
Youth +4
Fever Itching +3 +3
Uranium Tier -2 -2
=
11

Enough for 2 uranium at 2x or 1 at 3x.
OR
You could also add in self resp and the plasma stuff to have a more reasonable amount of healing.

Either way you don't have enough healing to pull you out of reasonable crit without uranium.

**Comment by kevinz000** [comment]
Fuck I thought https://github.com/tgstation/tgstation/pull/21029 was merged @Nabski89

**Comment by kevinz000** [comment]
hey xdtm go reopen 21029 :^)

**Comment by kevinz000** [comment]
@Nabski89 if it was it'd be +6 for stagespeed because you could just reroll but >>>people hate fun 
-_-

**Comment by Cyberboss** [comment]
Shh we know

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by XDTM** [comment]
fuck again

**Comment by XDTM** [comment]
never using command line again

**Pull request merged**
